File: Tutorials and Problems/P2c/Solutions c/LunchList3b/app/src/main/java/org/feup/apm/lunchlist3b/Restaurant.java
package org.feup.apm.lunchlist3b;

import androidx.annotation.NonNull;

import java.io.Serializable;

public class Restaurant implements Serializable {
  private String name="";
  private String address="";
  private String type="";
  private String notes="";

  public String getName() {
    return(name);
  }
  public void setName(String name) {
    this.name=name;
  }

  public String getAddress() {
    return(address);
  }
  public void setAddress(String address) {
    this.address=address;
  }

  public String getType() {
    return(type);
  }
  public void setType(String type) {
    this.type=type;
  }

  public String getNotes() {
    return(notes);
  }
  public void setNotes(String notes) {
    this.notes=notes;
  }

  public @NonNull
  String toString() {
    return(getName());
  }
}

File: Tutorials and Problems/3.3. The Lunch Restaurants App/LunchList3b/app/src/main/java/org/feup/apm/lunchlist3b/Restaurant.kt
package org.feup.apm.lunchlist3b

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import java.io.Serializable

// The class for a Restaurant data (refactored in Kotlin style with a primary constructor)

data class Restaurant(val name: String, val address: String, val type: String, val notes: String):
  Serializable {
  override fun toString(): String {
    return name
  }
}

class RestaurantAdapter(val ctx: Context, val rests: ArrayList<Restaurant>): ArrayAdapter<Restaurant>(ctx, R.layout.list_row, rests) {
  override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
    val row = convertView ?: (ctx as AppCompatActivity).layoutInflater.inflate(R.layout.list_row, parent, false)
    val r = rests[position]
    row.findViewById<TextView>(R.id.title).text = r.name
    row.findViewById<TextView>(R.id.address).text = r.address
    val symbol = row.findViewById<ImageView>(R.id.symbol)
    when (r.type) {
      "sit" -> symbol.setImageResource(R.drawable.ball_red)
      "take" -> symbol.setImageResource(R.drawable.ball_yellow)
      "delivery" -> symbol.setImageResource(R.drawable.ball_green)
    }
    return row
  }
}